### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by susodapop**
## Description

- Adds the e2e test suite
- Updates CONTRIBUTING with invocation instructions
- Updates the github action for unit tests for clarity

### CODE REVIEW

File: .github/workflows/code-quality-checks.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Code Quality Checks
 on: [push]
 jobs:
-  run-tests:
+  run-unit-tests:
```
Comment (moderakh) [inline .github/workflows/code-quality-checks.yml:4]: what does unit test mean in this context? 

Is this mock unit test without the need for databricks account?
or is it integration test e2e which require databricks account

**Review by yunbodeng-db — APPROVED** [review]

File: .github/workflows/code-quality-checks.yml (hunk 1 above)
Comment (susodapop) [reply .github/workflows/code-quality-checks.yml:4]: `unit` means no databricks account is required.
`e2e` means databricks account is required.

Comment (susodapop) [reply .github/workflows/code-quality-checks.yml:4]: I'm updating the CONTRIBUTING doc with this info.

Comment (moderakh) [reply .github/workflows/code-quality-checks.yml:4]: thanks Jesse.

**Pull request merged**